src/ImageSharp/Metadata/Profiles/XMP/XmpProfile.cs

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Xml.Linq;
+
+namespace SixLabors.ImageSharp.Metadata.Profiles.Xmp
+{
+    /// <summary>
+    /// Represents an XMP profile, providing access to the raw XML.
+    /// See <seealso href="https://www.adobe.com/devnet/xmp.html"/> for the full specification.
+    /// </summary>
+    public sealed class XmpProfile : IDeepCloneable<XmpProfile>
+    {
+        private XDocument document;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="XmpProfile"/> class.
+        /// </summary>
+        public XmpProfile()
+            : this((byte[])null)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="XmpProfile"/> class.
+        /// </summary>
+        /// <param name="data">The UTF8 encoded byte array to read the XMP profile from.</param>
+        public XmpProfile(byte[] data) => this.Data = data;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="XmpProfile"/> class, based on the speicief <see cref="XDocument"/>.
+        /// </summary>
+        /// <param name="doc">The document to base this instance on.</param>
+        public XmpProfile(XDocument doc)
+        {
+            this.document = doc;
+            this.UpdateData();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="XmpProfile"/> class
+        /// by making a copy from another XMP profile.
+        /// </summary>
+        /// <param name="other">The other XMP profile, from which the clone should be made from.</param>
+        private XmpProfile(XmpProfile other)
+        {
+            Guard.NotNull(other, nameof(other));
+
+            if (other.Data != null)
+            {
+                this.Data = new byte[other.Data.Length];
+                other.Data.AsSpan().CopyTo(this.Data);
+            }
+        }
+
+        /// <summary>
+        /// Gets the rax XML document containing the XMP profile.
+        /// </summary>
+        public XDocument Document
```

Comment:
`XDocument` is not read-only. A user adding nodes to the doc may expect their changes to be written back by re-encoding. We definitely don't want to do that, since we are not validating the contents.

My recommendation is to turn this into a `.GetDocument()` method that will create a new document on each call.

@JimBobSquarePants thoughts?

Reply:
The profile is supposed to be writeable no? If not I don't see a clear purpose for this. 

I'm not altogether happy with the lack of types and read/write functionality tbh. Our metadata profiles EXIF/XMP should follow a similar structure.

Reply:
From https://github.com/SixLabors/ImageSharp/issues/858#issuecomment-998446230:

> Creating an object model against XMP is IMO not worth the effort unless you find strong need for it.

Considering the format seems to be quite complicated, I can agree with this. I think preserving the data on load+resave has a value on it's own, we just need to make sure the types are extensible enabling to add actual contents.

Reply:
We already had preservation in Tiff using the raw byte array. I can go with preservation but it feels half baked to expose the thing in a readonly format while also expecting the user to actually parse the document.

Reply:
I think half-baked is fine, as long as: (1) users find it valuable, (2) the solution is extensible to make it fully baked.

I don't think we have the capacity to validate and review a PR adding full XMP support now.

Reply:
I was actually making some headway on trying to create an object model against XMP. It's far from complete, but I don't think it would be *too* complicated to have a user-friendly way to read and modify XMP. Would probably still be best if it is (potentially) added in a later PR though, and this PR just exposes the `XDocument`.

Reply:
Point 1 is the important one for V2. I would expect us to deliver a full mutable API for V3. 

`.GetDocument()` would definitely be better API-wise. I also don't believe we should expose the raw `Data` property and use `ToByteArray()` like our `ExifProfile`.

Reply:
Ok, sounds like a plan.

The `ToByteArray` method return a clone, as opposed to the `Data` property previously. The `ExifProfile` does this implicitly, as it parses its internal strong types.

The `GetDocument` method does no longer influence the original byte array.

And last but not least, the equality methods and operators has been removed.

@JimBobSquarePants This would then be the V2 API, as you describe.
@jubilant-enigma Feel free to implement your object model on top of this PR.